sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala

```diff
@@ -41,10 +41,7 @@ abstract class SubqueryExpression extends LeafExpression {
   /** Updates the query with new logical plan. */
   def withNewPlan(plan: LogicalPlan): SubqueryExpression
 
-  /** The correlated conditions that connects the sub-query to its parent query. */
-  def conditions: Seq[Expression]
-
-  protected def conditionString: String = conditions.mkString("[", " && ", "]")
+  protected def conditionString: String = children.mkString("[", " && ", "]")
```

Comment:
We can. I just put it there because I need it for correlated scalar subqueries (next PR)...